### PR TITLE
Get entity with specific role/group from tracker

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Update the method `get_latest_entity_values` to take an optional role/group label. Needed to allow users to fill a slot that depends on a specific role/group in a custom action.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/core/forms.rst [hunk 1]
```diff
@@ -142,10 +142,11 @@ Here's an example for the restaurant bot:
 
 The predefined functions work as follows:
 
-- ``self.from_entity(entity=entity_name, intent=intent_name)``
+- ``self.from_entity(entity=entity_name, intent=intent_name, role=role_name, group=group_name)``
   will look for an entity called ``entity_name`` to fill a slot
   ``slot_name`` regardless of user intent if ``intent_name`` is ``None``
-  else only if the users intent is ``intent_name``.
+  else only if the users intent is ``intent_name``. If ``role_name`` and/or ``group_name``
+  are/is provided, the role/group label of the entity also needs to match the given values.
```
Comment (ricwo) [inline docs/core/forms.rst:149]:
```suggestion
  are provided, the role/group label of the entity also needs to match the given values.
```

File: changelog/3765.feature.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Add support for composite entities in Rasa NLU.
+
+Composite entities allow you to define a role and/or group label in addition to the entity type.
+For more details see :ref:`composite-entities`.
```
Comment (ricwo) [inline changelog/3765.feature.rst:4]: I'd add add one sentence about what role and group labels are

File: rasa/core/trackers.py [hunk 3]
```diff
@@ -230,17 +236,34 @@ def get_slot(self, key: Text) -> Optional[Any]:
...
-        If no entity is found `None` is the default result."""
+        If no entity is found `None` is the default result.
+
+        Args:
+            entity_type: the entity type of interest
+            entity_role: optional entity role of interest
+            entity_group: optional entity group of interest
+
+        Returns:
+            List of entity values.
```
Comment (ricwo) [inline rasa/core/trackers.py:258]:
```suggestion
            Entity values.
```

File: tests/core/test_trackers.py [hunk 4]
```diff
@@ -206,29 +206,68 @@ def test_tracker_store(store, pair):
...
+        (
+            [
+                {"value": "greet", "entity": "entity_name"},
+                {"value": "bye", "entity": "entity_name", "group": "group"},
+            ],
+            ["greet", "bye"],
+        ),
+    ],
+)
+def test_tracker_entity_retrieval(entities, expected_value, default_domain: Domain):
```
Comment (ricwo) [inline tests/core/test_trackers.py:250]: please add types for `entities` and `expected_values`

File: tests/core/test_trackers.py [hunk 5]
```diff
@@ -206,29 +206,68 @@ def test_tracker_store(store, pair):
     assert [e.type_name for e in tracker.events] == expected
 
 
-def test_tracker_entity_retrieval(default_domain: Domain):
+@pytest.mark.parametrize(
+    "entities, expected_value",
```
Comment (ricwo) [inline tests/core/test_trackers.py:210]:
```suggestion
    "entities, expected_values",
```
as it's a list

File: tests/core/test_trackers.py (hunk 4 above)
Comment (ricwo) [inline tests/core/test_trackers.py:250]:
```suggestion
def test_tracker_entity_retrieval(entities, expected_values, default_domain: Domain):
```

File: tests/core/test_trackers.py [hunk 6]
```diff
@@ -206,29 +206,68 @@ def test_tracker_store(store, pair):
...
-            ],
+    tracker.update(UserUttered("/greet", intent, entities))
+
+    assert (
+        list(
+            tracker.get_latest_entity_values(
+                entity_type, entity_role=entity_role, entity_group=entity_group
+            )
         )
+        == expected_value
```
Comment (ricwo) [inline tests/core/test_trackers.py:269]:
```suggestion
        == expected_values
```

File: tests/core/test_trackers.py (hunk 4 above)
Comment (ricwo) [inline tests/core/test_trackers.py:250]:
```suggestion
def test_get_latest_entity_values(entities, expected_value, default_domain: Domain):
```

File: changelog/3765.feature.rst [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+To fill slots from entities with a specific role/group, you need to either use forms or use a custom action.
+We updated the tracker method ``get_latest_entity_values`` to take an optional role/group label.
```
[the remark this answers is not in the document]
Comment (tabergma) [reply changelog/3765.feature.rst:7]: Yes, PR is already open https://github.com/RasaHQ/rasa-sdk/pull/163

**Pull request merged**